src/inmanta/server/services/compilerservice.py

```diff
@@ -213,37 +218,73 @@ async def run(self, force_update: Optional[bool] = False) -> Tuple[bool, Optiona
                 await self._info("Creating project directory for environment %s at %s" % (environment_id, project_dir))
                 os.mkdir(project_dir)
 
-            repo_url: str = env.repo_url
-            repo_branch: str = env.repo_branch
-            if not repo_url:
-                if not os.path.exists(os.path.join(project_dir, "project.yml")):
-                    await self._warning(f"Failed to compile: no project found in {project_dir} and no repository set")
-                await self._end_stage(0)
-            else:
-                await self._end_stage(0)
-                # checkout repo
+            async def update_modules() -> data.Report:
+                LOGGER.info("Updating modules")
+                return await self._run_compile_stage(
+                    "Updating modules", [*inmanta_path, "-vvv", "-X", "modules", "update"], project_dir
+                )
+
+            async def install_modules() -> data.Report:
+                LOGGER.info("Installing modules")
+                return await self._run_compile_stage(
+                    "Installing modules", [*inmanta_path, "-vvv", "-X", "project", "install"], project_dir
+                )
+
+            async def setup() -> AsyncIterator[Awaitable[data.Report]]:
```

Comment:
I noticed that this code assumes that the `repo_url` never changes after the initial clone operation. That looks like a bug to me.

Reply:
See my comment on slack.